### PR TITLE
Introduce Dir helper for scrollLeft

### PR DESCRIPTION
**Pull request by yuriy-fix**
Connected to https://github.com/vaadin/vaadin-tabs/issues/106

### CODE REVIEW

File: vaadin-dir-helper.html [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  window.Vaadin = window.Vaadin || {};
+
+  Vaadin.DirHelper = class VaadinDirHelper {
+    /**
+     * Get the scroll type in the current browser view.
+     *
+     * @return {String} the scroll type. Possible values are `default|reverse|negative`
+     */
+    static _detectScrollType() {
+      const inDOM = !!(typeof window !== 'undefined' && window.document && window.document.createElement);
```
Comment (web-padawan) [inline vaadin-dir-helper.html]: Seems like this check is for node.js so you can remove it completely.

File: vaadin-dir-helper.html [hunk 2]
```diff
@@ -0,0 +1,97 @@
+<!--
+@license
+Copyright (c) 2020 Vaadin Ltd.
+This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+-->
+
+<script>
+  window.Vaadin = window.Vaadin || {};
+
+  Vaadin.DirHelper = class VaadinDirHelper {
```
Comment (web-padawan) [inline vaadin-dir-helper.html]: I'm not sure but this potentially could leak reference to `VaadinDirHelper` class globally.

Please consider wrapping into IIFE same as we do here in [Vaadin.FocusablesHelper](https://github.com/vaadin/vaadin-overlay/blob/master/src/vaadin-focusables-helper.html)

File: vaadin-dir-helper.html [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    /**
+     * Get the scroll type in the current browser view.
+     *
+     * @return {String} the scroll type. Possible values are `default|reverse|negative`
+     */
+    static _detectScrollType() {
```
Comment (web-padawan) [inline vaadin-dir-helper.html]: You don't really need a class here, neither `static`. Helper can be a plain object with a method.

Also, this method should be considered "public" (called from outside) and not underscored.
Let's change this to be aligned with [Vaadin.FocusablesHelper](https://github.com/vaadin/vaadin-overlay/blob/master/src/vaadin-focusables-helper.html) approach.

File: vaadin-dir-helper.html [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      const inDOM = !!(typeof window !== 'undefined' && window.document && window.document.createElement);
+      if (!inDOM || !window.document.body) {
+        return 'indeterminate';
+      }
+
+      var dummy = window.document.createElement('div');
```
Comment (web-padawan) [inline vaadin-dir-helper.html]: nit: use `const` and remove `window`

File: vaadin-dir-helper.html [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      dummy.appendChild(document.createTextNode('ABCD'));
```
Comment (web-padawan) [inline vaadin-dir-helper.html]: nit: you can set `textContent` instead of creating a text node.

File: vaadin-dir-helper.html [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    /**
+     * Get the scrollLeft value of the element relative to the direction
+     *
+     * @param {String} scrollType type of the scroll detected with `_detectScrollType`
+     * @param {String} direction current direction of the element
+     * @param {Node} element
+     * @return {Number} the scrollLeft value.
+     */
+    static getNormalizedScrollLeft(scrollType, direction, element) {
+      var scrollLeft = element.scrollLeft;
```
Comment (web-padawan) [inline vaadin-dir-helper.html]: nit: use `const`. 

Also you can use destructuring: `const {scrollLeft} = element;`

File: vaadin-dir-helper.html [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+          break;
+        case 'reverse':
+          element.scrollLeft = element.scrollWidth - element.clientWidth - scrollLeft;
+          break;
+        default:
+          element.scrollLeft = scrollLeft;
+          break;
+      }
+    }
+
```
Comment (web-padawan) [inline vaadin-dir-helper.html]: nit: extra newline

**Review by web-padawan — CHANGES_REQUESTED** [review]
Requested some changes, related to code style only.

It would be nice to also add a mention of specific browsers (including IE11) that need this.

File: vaadin-dir-helper.html (hunk 1 above)
Comment (yuriy-fix) [reply vaadin-dir-helper.html]: Removed.

File: vaadin-dir-helper.html (hunk 2 above)
Comment (yuriy-fix) [reply vaadin-dir-helper.html]: Updated.

File: vaadin-dir-helper.html (hunk 3 above)
Comment (yuriy-fix) [reply vaadin-dir-helper.html]: Updated.

File: vaadin-dir-helper.html (hunk 4 above)
Comment (yuriy-fix) [reply vaadin-dir-helper.html]: Updated.

File: vaadin-dir-helper.html (hunk 6 above)
Comment (yuriy-fix) [reply vaadin-dir-helper.html]: Updated.

File: vaadin-dir-helper.html (hunk 7 above)
Comment (yuriy-fix) [reply vaadin-dir-helper.html]: Removed.

File: vaadin-dir-helper.html (hunk 5 above)
Comment (yuriy-fix) [reply vaadin-dir-helper.html]: Updated

**Review by web-padawan — APPROVED** [review]

**Pull request merged**